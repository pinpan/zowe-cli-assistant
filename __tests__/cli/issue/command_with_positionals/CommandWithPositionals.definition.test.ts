/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { relative, sep, posix } from "path";

describe("Command With Positionals definition", () => {
    it("should match the snapshot", () => {
        const contents = require("../../../../src/cli/issue/command_with_positionals/CommandWithPositionals.definition");
        Object.keys(contents).forEach(key => contents[key].handler = relative(__dirname, contents[key].handler).split(sep).join(posix.sep));
        expect(contents).toMatchSnapshot();
    });
});
