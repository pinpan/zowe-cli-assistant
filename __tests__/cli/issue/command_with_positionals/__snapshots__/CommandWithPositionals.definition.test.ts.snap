// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Command With Positionals definition should match the snapshot 1`] = `
Object {
  "CommandWithPositionalsDefinition": Object {
    "aliases": Array [
      "cwp",
    ],
    "description": "[objects] in Zowe CLI are the entities on which [actions] are perfomed. [objects] are always nouns. For example, for command \\"zowe zos-files delete data-set\\", the [object] is \\"data-set\\". For this command, we will simply print console error (stderr) messages and the command should succeed.",
    "handler": "../../../../src/cli/issue/command_with_positionals/CommandWithPositionals.handler",
    "name": "command-with-positionals",
    "positionals": Array [
      Object {
        "description": "The first positional parameter",
        "name": "firstpositional",
        "required": true,
        "type": "string",
      },
      Object {
        "description": "The second positional parameter",
        "name": "secondpositional",
        "required": true,
        "type": "number",
      },
    ],
    "summary": "[object] command-with-positionals prints console msgs",
    "type": "command",
  },
}
`;
